tincr/io/library/genCellLibrary.tcl

```diff
@@ -291,6 +291,110 @@ proc processMacroCell {c s fo} {
     }
 }
 
+proc doPorts { fo } {
+
+    puts "Doing ports..."
+
+    set lc [get_lib_cells]
```

Comment:
This information should be cached so that multiple calls to the same method don't duplicate work. In fact, the `lib_cells compatible_with` method already does this...